File: myonlinefm/src/main/java/com/sec/myonlinefm/classificationprogram/RequestProgramClassifyActivity.java
package com.sec.myonlinefm.classificationprogram;

import android.content.Intent;
import android.support.v4.app.FragmentTransaction;
import android.support.annotation.NonNull;
import android.support.v7.app.ActionBar;
import android.content.ComponentName;
import android.content.ServiceConnection;
import android.os.IBinder;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;
import android.widget.SearchView;
import android.widget.TextView;

import com.sec.myonlinefm.OnLineFMPlayerService;
import com.sec.myonlinefm.R;
import com.sec.myonlinefm.SearchActivity;
import com.sec.myonlinefm.classificationprogram.mainfacefragment.OnLineFavFragment;
import com.sec.myonlinefm.classificationprogram.mainfacefragment.OnLineHomeFragment;
import com.sec.myonlinefm.dbdata.MySQLHelper;

public class RequestProgramClassifyActivity extends AppCompatActivity implements ServiceConnection , View.OnClickListener{

    private static OnLineFMPlayerService mService;

    private View topBarActionBarView = null;

    private String TAG = "RequestProgramClassifyActivity";
    private String mParam1 = TAG;
    private String mParam2;

    private OnLineHomeFragment home;
    private OnLineFavFragment favFragment;

    private String mCurrentFragmentTag;

    private void setTopPanelOnActionBar(String title, boolean isShowSearchView) {
        ActionBar bar = getSupportActionBar();
        if (bar != null) {
            bar.setHomeButtonEnabled(true);
            bar.setDisplayHomeAsUpEnabled(false);
            bar.setDisplayShowHomeEnabled(false);
            bar.setDisplayShowTitleEnabled(false);
            bar.setBackgroundDrawable(getDrawable(R.drawable.titlebar_bg));
            bar.setElevation(0);
        }
        topBarActionBarView = getLayoutInflater().inflate(R.layout.top_bar_panel_online, null);
        ImageView mBackBut = topBarActionBarView.findViewById(R.id.back_but);
        mBackBut.setVisibility(View.INVISIBLE);
        TextView mTopBarTitle = topBarActionBarView.findViewById(R.id.top_bar_title);
        SearchView mOnLineSearchView = topBarActionBarView.findViewById(R.id.online_search);
        mOnLineSearchView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(RequestProgramClassifyActivity.this, SearchActivity.class);
                startActivity(i);
            }
        });
        mTopBarTitle.setText(title);
        if(isShowSearchView) {
            mTopBarTitle.setVisibility(View.GONE);
            mOnLineSearchView.setVisibility(View.VISIBLE);
        } else {
            mTopBarTitle.setVisibility(View.VISIBLE);
            mOnLineSearchView.setVisibility(View.GONE);
        }
        if (bar != null) {
            bar.setCustomView(topBarActionBarView);
            bar.setDisplayHomeAsUpEnabled(true);
            bar.setDisplayOptions(ActionBar.DISPLAY_SHOW_CUSTOM);
            bar.show();
        }
    }

    @Override
    public void onSupportActionModeStarted(@NonNull android.support.v7.view.ActionMode mode) {
        super.onSupportActionModeStarted(mode);
        if (topBarActionBarView != null)
            topBarActionBarView.setVisibility(View.VISIBLE);
    }

    @Override
    public void onSupportActionModeFinished(@NonNull android.support.v7.view.ActionMode mode) {
        super.onSupportActionModeFinished(mode);
        if (topBarActionBarView != null)
            topBarActionBarView.setVisibility(View.VISIBLE);
    }

    private FragmentTransaction ft;
    private ImageView mFavIcon;
    private ImageView mHomeIcon;
    private ImageView mDownLoadIcon;
    private ImageView mUserIcon;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_request_program_classify);
        MySQLHelper mySQLHelper = new MySQLHelper(this, "onLineFM.db", null, 1);

        ft = getSupportFragmentManager().beginTransaction();
        home = (OnLineHomeFragment)getSupportFragmentManager().findFragmentByTag(OnLineHomeFragment.TAG);
        mCurrentFragmentTag = OnLineHomeFragment.TAG;
        if(home == null) {
            mParam2 = "home";
            ft.add(OnLineHomeFragment.newInstance(mParam1, mParam2), OnLineHomeFragment.TAG);
        } else {
            ft.attach(home);
        }
        ft.commitAllowingStateLoss();

        mFavIcon = (ImageView) findViewById(R.id.into_fave_icon);
        mHomeIcon = (ImageView) findViewById(R.id.home);
        mDownLoadIcon = (ImageView) findViewById(R.id.down_load_file);
        mUserIcon = (ImageView) findViewById(R.id.user_info);

        mFavIcon.setClickable(true);
        mFavIcon.setOnClickListener(this);

        mHomeIcon.setClickable(true);
        mHomeIcon.setOnClickListener(this);

        mDownLoadIcon.setClickable(true);
        mDownLoadIcon.setOnClickListener(this);

        mUserIcon.setClickable(true);
        mUserIcon.setOnClickListener(this);

        mService = OnLineFMPlayerService.getMyservice();

        upDateActionBarView ();
        upDateImageTint();
    }

    public void upDateActionBarView () {
        switch (mCurrentFragmentTag) {
            case OnLineHomeFragment.TAG:
                setTopPanelOnActionBar("NetRadio", true);
                break;
            case OnLineFavFragment.TAG:
                setTopPanelOnActionBar("我的收藏", false);
                break;
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (topBarActionBarView != null)
            topBarActionBarView.setVisibility(View.VISIBLE);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    /* connect to service
    * **/
    @Override
    public void onServiceConnected(ComponentName name, IBinder service) {
        if(service instanceof OnLineFMPlayerService.ServiceBinder){
            OnLineFMPlayerService.ServiceBinder binder = (OnLineFMPlayerService.ServiceBinder)service;
            binder.getService();
        }
    }
    @Override
    public void onServiceDisconnected(ComponentName name) {

    }

    private int mCurrentMainFragment = R.id.home;

    void upDateImageTint() {
        switch (mCurrentMainFragment) {
            case R.id.home:
                mHomeIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_b));
                mFavIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mDownLoadIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mUserIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                break;
            case R.id.into_fave_icon:
                mHomeIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mFavIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_b));
                mDownLoadIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mUserIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                break;
            case R.id.down_load_file:
                mHomeIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mFavIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mDownLoadIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_b));
                mUserIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                break;
            case R.id.user_info:
                mHomeIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mFavIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mDownLoadIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_a));
                mUserIcon.setColorFilter(getResources().getColor(R.color.net_fm_back_b));
                break;
        }
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.into_fave_icon:
                favFragment = (OnLineFavFragment)getSupportFragmentManager().findFragmentByTag(OnLineFavFragment.TAG);
                mCurrentFragmentTag = OnLineFavFragment.TAG;
                if(favFragment == null) {
                    mParam2 = "favFragment";
                    favFragment = OnLineFavFragment.newInstance(mParam1, mParam2);
                }
                ft = getSupportFragmentManager().beginTransaction();
                ft.replace(R.id.main_online_fragment, favFragment);
                ft.commit();
                mCurrentMainFragment = R.id.into_fave_icon;
                break;
            case R.id.home:
                mCurrentFragmentTag = OnLineHomeFragment.TAG;
                home = (OnLineHomeFragment)getSupportFragmentManager().findFragmentByTag(OnLineHomeFragment.TAG);
                if(home == null) {
                    mParam2 = "home";
                    home = OnLineHomeFragment.newInstance(mParam1, mParam2);
                }
                ft = getSupportFragmentManager().beginTransaction();
                ft.replace(R.id.main_online_fragment, home);
                ft.commit();
                mCurrentMainFragment = R.id.home;
                break;
        }
        upDateActionBarView ();
        upDateImageTint();
    }
}

File: myonlinefm/src/main/java/com/sec/myonlinefm/classificationprogram/RequestProgramClassifyActivity.kt
package com.sec.myonlinefm.classificationprogram

import android.view.*
import android.widget.*
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.view.ActionMode
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction

import android.os.Bundle

import android.content.Intent
import com.sec.myonlinefm.classificationprogram.mainfacefragment.OnLineFavFragment

import com.sec.myonlinefm.classificationprogram.mainfacefragment.OnLineHomeFragment
import android.content.ServiceConnection
import android.content.ComponentName
import android.os.IBinder
import com.sec.myonlinefm.R
import com.sec.myonlinefm.dbdata.MySQLHelper
import com.sec.myonlinefm.OnLineFMPlayerService
import com.sec.myonlinefm.SearchActivity

class RequestProgramClassifyActivity : AppCompatActivity(), ServiceConnection, View.OnClickListener {
    private var topBarActionBarView: View? = null
    private val TAG: String? = "RequestProgramClassifyActivity"
    private val mParam1 = TAG
    private var mParam2: String? = null
    private var home: Fragment? = null
    private var favFragment: Fragment? = null
    private var mCurrentFragmentTag: String? = null
    private fun setTopPanelOnActionBar(title: String?, isShowSearchView: Boolean) {
        val bar = supportActionBar
        if (bar != null) {
            bar.setHomeButtonEnabled(true)
            bar.setDisplayHomeAsUpEnabled(false)
            bar.setDisplayShowHomeEnabled(false)
            bar.setDisplayShowTitleEnabled(false)
            bar.setBackgroundDrawable(getDrawable(R.drawable.titlebar_bg))
            bar.elevation = 0f
        }
        topBarActionBarView = layoutInflater.inflate(R.layout.top_bar_panel_online, null)
        val mBackBut = topBarActionBarView!!.findViewById<ImageView?>(R.id.back_but)
        mBackBut.visibility = View.INVISIBLE
        val mTopBarTitle = topBarActionBarView!!.findViewById<TextView?>(R.id.top_bar_title)
        val mOnLineSearchView = topBarActionBarView!!.findViewById<SearchView?>(R.id.online_search)
        mOnLineSearchView.setOnClickListener {
            val i = Intent(this@RequestProgramClassifyActivity, SearchActivity::class.java)
            startActivity(i)
        }
        mTopBarTitle.text = title
        if (isShowSearchView) {
            mTopBarTitle.visibility = View.GONE
            mOnLineSearchView.visibility = View.VISIBLE
        } else {
            mTopBarTitle.visibility = View.VISIBLE
            mOnLineSearchView.visibility = View.GONE
        }
        if (bar != null) {
            bar.customView = topBarActionBarView
            bar.setDisplayHomeAsUpEnabled(true)
            bar.displayOptions = ActionBar.DISPLAY_SHOW_CUSTOM
            bar.show()
        }
    }

    override fun onSupportActionModeStarted(mode: ActionMode) {
        super.onSupportActionModeStarted(mode)
        if (topBarActionBarView != null) topBarActionBarView!!.setVisibility(View.VISIBLE)
    }

    override fun onSupportActionModeFinished(mode: ActionMode) {
        super.onSupportActionModeFinished(mode)
        if (topBarActionBarView != null) topBarActionBarView!!.setVisibility(View.VISIBLE)
    }

    private var ft: FragmentTransaction? = null
    private var mFavIcon: ImageView? = null
    private var mHomeIcon: ImageView? = null
    private var mDownLoadIcon: ImageView? = null
    private var mUserIcon: ImageView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_request_program_classify)
        val mySQLHelper = MySQLHelper(this, "onLineFM.db", null, 1)
        ft = supportFragmentManager.beginTransaction()
        home = supportFragmentManager.findFragmentByTag(OnLineHomeFragment.Companion.TAG) as OnLineHomeFragment?
        mCurrentFragmentTag = OnLineHomeFragment.Companion.TAG
        if (home == null) {
            mParam2 = "home"
            ft!!.add(OnLineHomeFragment.Companion.newInstance(mParam1, mParam2)!!, OnLineHomeFragment.Companion.TAG)
        } else {
            ft!!.attach(home!!)
        }
        ft!!.commitAllowingStateLoss()
        mFavIcon = findViewById<View?>(R.id.into_fave_icon) as ImageView
        mHomeIcon = findViewById<View?>(R.id.home) as ImageView
        mDownLoadIcon = findViewById<View?>(R.id.down_load_file) as ImageView
        mUserIcon = findViewById<View?>(R.id.user_info) as ImageView
        mFavIcon!!.setClickable(true)
        mFavIcon!!.setOnClickListener(this)
        mHomeIcon!!.setClickable(true)
        mHomeIcon!!.setOnClickListener(this)
        mDownLoadIcon!!.setClickable(true)
        mDownLoadIcon!!.setOnClickListener(this)
        mUserIcon!!.setClickable(true)
        mUserIcon!!.setOnClickListener(this)
        mService = OnLineFMPlayerService.Companion.getMyservice()
        upDateActionBarView()
        upDateImageTint()
    }

    fun upDateActionBarView() {
        when (mCurrentFragmentTag) {
            OnLineHomeFragment.Companion.TAG -> setTopPanelOnActionBar("NetRadio", true)
            OnLineFavFragment.Companion.TAG -> setTopPanelOnActionBar("我的收藏", false)
        }
    }

    override fun onResume() {
        super.onResume()
        if (topBarActionBarView != null) topBarActionBarView!!.setVisibility(View.VISIBLE)
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    /* connect to service
    * **/
    override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
        if (service is OnLineFMPlayerService.ServiceBinder) {
            val binder = service as OnLineFMPlayerService.ServiceBinder?
            binder!!.getService()
        }
    }

    override fun onServiceDisconnected(name: ComponentName?) {}
    private var mCurrentMainFragment = R.id.home
    fun upDateImageTint() {
        when (mCurrentMainFragment) {
            R.id.home -> {
                mHomeIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_b))
                mFavIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mDownLoadIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mUserIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
            }
            R.id.into_fave_icon -> {
                mHomeIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mFavIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_b))
                mDownLoadIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mUserIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
            }
            R.id.down_load_file -> {
                mHomeIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mFavIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mDownLoadIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_b))
                mUserIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
            }
            R.id.user_info -> {
                mHomeIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mFavIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mDownLoadIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_a))
                mUserIcon!!.setColorFilter(resources.getColor(R.color.net_fm_back_b))
            }
        }
    }

    override fun onClick(v: View?) {
        when (v!!.getId()) {
            R.id.into_fave_icon -> {
                favFragment = supportFragmentManager.findFragmentByTag(OnLineFavFragment.Companion.TAG)
                mCurrentFragmentTag = OnLineFavFragment.Companion.TAG
                if (favFragment == null) {
                    mParam2 = "favFragment"
                    favFragment = OnLineFavFragment.Companion.newInstance(mParam1, mParam2)
                }
                ft = supportFragmentManager.beginTransaction()
                ft!!.replace(R.id.main_online_fragment, favFragment!!)
                ft!!.commit()
                mCurrentMainFragment = R.id.into_fave_icon
            }
            R.id.home -> {
                mCurrentFragmentTag = OnLineHomeFragment.Companion.TAG
                home = supportFragmentManager.findFragmentByTag(OnLineHomeFragment.Companion.TAG) as OnLineHomeFragment?
                if (home == null) {
                    mParam2 = "home"
                    home = OnLineHomeFragment.Companion.newInstance(mParam1, mParam2)
                }
                ft = supportFragmentManager.beginTransaction()
                ft!!.replace(R.id.main_online_fragment, home!!)
                ft!!.commit()
                mCurrentMainFragment = R.id.home
            }
        }
        upDateActionBarView()
        upDateImageTint()
    }

    companion object {
        private var mService: OnLineFMPlayerService? = null
    }
}